### PR TITLE
Frontend: Set moduleResolution to bundler in TSconfig

### PR DESCRIPTION
**Pull request by jackw**
**What is this feature?**

This PR changes the TS config to use `moduleResolution: "bundler"`.

**Why do we need this feature?**

So we can use the new and shiny in our typescript.

>'bundler' for use with bundlers. Like node16 and nodenext, this mode supports package.json "imports" and "exports", but unlike the Node.js resolution modes, bundler never requires file extensions on relative paths in imports.

**Who is this feature for?**

Grafana contributors.

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

File: .yarn/patches/@prometheus-io-lezer-promql-npm-0.55.1-60d9541361.patch
Comment (ashharrison90) [inline .yarn/patches/@prometheus-io-lezer-promql-npm-0.55.1-60d9541361.patch:1]: i guess this patch goes away once the upstream PRs are merged? 🤔

**Review by ashharrison90 — COMMENTED** [review]
oooo shiny 😍 

won't approve yet, but lgtm 👍

**Review by aangelisc — APPROVED** [review]
LGTM from Partner DS!

File: .yarn/patches/@prometheus-io-lezer-promql-npm-0.55.1-60d9541361.patch
Comment (jackw) [reply .yarn/patches/@prometheus-io-lezer-promql-npm-0.55.1-60d9541361.patch:1]: Exactly. 👍 

I'll update the lib in this PR once it's gone out the door.

File: public/app/plugins/datasource/azuremonitor/tsconfig.json
Comment (jackw) [inline public/app/plugins/datasource/azuremonitor/tsconfig.json:4]: This isn't necessary for this PR but testing led me to discover types errors in my IDE for all the azuremonitor tests that use testing-library assertions.

File: public/app/plugins/datasource/cloud-monitoring/tsconfig.json
Comment (jackw) [inline public/app/plugins/datasource/cloud-monitoring/tsconfig.json:4]: Again, not necessary for this PR but testing led me to discover types errors in my IDE for all the cloud-monitoring tests that use testing-library assertions.

File: .yarn/patches/@prometheus-io-lezer-promql-npm-0.55.1-60d9541361.patch
Comment (jackw) [reply .yarn/patches/@prometheus-io-lezer-promql-npm-0.55.1-60d9541361.patch:1]: Patch has been deleted as we moved to 0.300 of lezer-promql in [this PR](https://github.com/grafana/grafana/pull/96532).

**Review by leventebalogh — APPROVED** [review]
LGTM 👍

**Review by joshhunt — APPROVED** [review]
Looking good!

**Pull request merged**
